### ISSUE TITLE
A function for computing the minimum distance between two dihedral angles

### ISSUE THREAD

**Issue by xiki-tempula** [scipy/scipy#13673]
**Is your feature request related to a problem? Please describe.**
It is quite common that one might want to know the absolute difference between two dihedral angles or between two dihedral angle n-dimensional arrays, so it would be nice if such a function exists in scipy.stats similar to scipy.stats.circmean or scipy.stats.circvar.

**Describe the solution you'd like**

```
def circdelta(sample1, sample2, high=2*pi, low=0, nan_policy='propagate'):
    """
    Compute the circular difference between sample1 and sample2 in a range.
    Parameters
    ----------
    samples1 : array_like
        Input array 1.
    samples2 : array_like
        Input array 2.
    high : float or int, optional
        High boundary for circular delta range.  Default is ``2*pi``.
    low : float or int, optional
        Low boundary for circular delta range.  Default is 0.
    nan_policy : {'propagate', 'raise', 'omit'}, optional
        Defines how to handle when input contains nan. 'propagate' returns nan,
        'raise' throws an error, 'omit' performs the calculations ignoring nan
        values. Default is 'propagate'.
    Returns
    -------
    circdelta : array_like
        Circular delta.
    Examples
    --------
    >>> circdelta(0.1, 5)
    1.38
    >>> circdelta([0.1, 2], [5, 3])
    [1.38, 1]
    """
```

**Comment by tupui** [comment]
Hi! Are you proposing to implement this feature yourself?

**Comment by xiki-tempula** [comment]
Hi @tupui, 
I was wondering if such a function exist?
I think It would be nice if scipy could implement this function.

**Comment by tupui** [comment]
I don't think it exists no. You could try to ask on the mail list or if you're confident this is actually wanted propose a PR.

**Comment by V0lantis** [comment]
I am not very aware of the subject. Could you share some papers / formula which explains how to compute this difference?

**Comment by xiki-tempula** [comment]
@V0lantis It is just the minimum image convention.
A naive implementation would be:

def circdelta(sample1, sample2, high=2*pi, low=0, nan_policy='propagate'):
    sample1_treated = sample1 % (high - low)
    sample2_treated = sample2 % (high - low)
    return min(abs(sample1_treated-sample2_treated), abs(sample1_treated-sample2_treated+(high - low)), abs(sample1_treated-sample2_treated-(high - low)))

**Comment by V0lantis** [comment]
Ok thank you, I found a lot of interesting articles :-). 
Is this [one](https://www.researchgate.net/profile/Ulrich-Deiters/publication/235933545_Efficient_Coding_of_the_Minimum_Image_Convention/links/5955135d458515bbaa21e73e/Efficient-Coding-of-the-Minimum-Image-Convention.pdf?_sg%5B0%5D=g_p6uEzaivtl-WkNIVp_CblMt3jrOLMdp-EOdTYZDOp2Uys1_BK_ZoVmP9BuDM1WqbE9npJLUROukrkj9D0wZA.WmdNUum5jDq3ua6qgsvgeOEa6Q5IctSO8koWg6fD33_14ZAQ9I3wPW0vkqYrAEtgJrT7uDET4x7VSaciHnNzYA&_sg%5B1%5D=4sI6kJYzgVKxfe_l_P9rV4scj_xF4XaAXDOMF5STT2zcEg9foCzwtdHmdpNdXnxIZory5C8fjRAc5ITwHwQNwfJox__uP5CsP1_IYnEoq8YE.WmdNUum5jDq3ua6qgsvgeOEa6Q5IctSO8koWg6fD33_14ZAQ9I3wPW0vkqYrAEtgJrT7uDET4x7VSaciHnNzYA&_iepl=) a relevant one ?

**Comment by xiki-tempula** [comment]
@V0lantis Yes, it looks like a reasonable one to me. Thank you.

**Comment by mdhaber** [comment]
I'm not sure that `scipy.stats` is the most appropriate place for this as described, as it seems like elementwise subtraction on a circular domain. (If there were, for instance, a paired-sample hypothesis test to be performed, I would see the connection.) If you're interested in seeing this added, please send an email to the [mailing list](https://mail.python.org/mailman3/lists/scipy-dev.python.org/) explaining why this is fundamental, useful across multiple domains, and important to add despite the simple implementation in terms of existing functions. I'd suggest including well-cited references as support, and please open the discussion about where it would belong.
